src/Pomelo.EntityFrameworkCore.MySql/Update/Internal/MySqlBatchExecutor.cs

```diff
@@ -4,120 +4,84 @@
-        ///     directly from your code. This API may change or be removed in future releases.
-        /// </summary>
-        public virtual int Execute(
-            IEnumerable<ModificationCommandBatch> commandBatches,
-            IRelationalConnection connection)
+      var rowsAffected = 0;
+      connection.Open();
+      IDbContextTransaction startedTransaction = null;
+      try
+      {
+        if (connection.CurrentTransaction == null)
         {
-            var rowsAffected = 0;
-            connection.Open();
-            IDbContextTransaction startedTransaction = null;
-            try
-            {
-                if (connection.CurrentTransaction == null)
-                {
-                    startedTransaction = connection.BeginTransaction();
-                }
+          startedTransaction = connection.BeginTransaction();
+        }
 
-                foreach (var commandbatch in commandBatches)
-                {
-                    commandbatch.Execute(connection);
-                    // Fixed Issue #1: DataReader conflicted when added multiple entities
-                    try
-                    {
-                        lock (this)
-                        {
-                            var reader = (connection.DbConnection as MySqlConnection).Reader.LastOrDefault();
-                            if (reader != null)
-                            {
-                                reader.Dispose();
-                                (connection.DbConnection as MySqlConnection).Reader.Remove(reader);
-                            }
-                        }
-                    }
-                    catch
-                    {
-                    }
-                    rowsAffected += commandbatch.ModificationCommands.Count;
-                }
+        foreach (var commandbatch in commandBatches)
+        {
+            commandbatch.Execute(connection);
+            rowsAffected += commandbatch.ModificationCommands.Count;
+        }
+        startedTransaction?.Commit();
+      }
+      finally
+      {
+        startedTransaction?.Dispose();
+        connection.Close();
+      }
 
-                startedTransaction?.Commit();
-            }
-            finally
-            {
-                startedTransaction?.Dispose();
-                connection.Close();
-            }
+      return rowsAffected;
+    }
 
-            return rowsAffected;
+    /// <summary>
+    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    public virtual async Task<int> ExecuteAsync(
+      IEnumerable<ModificationCommandBatch> commandBatches,
+      IRelationalConnection connection,
+      CancellationToken cancellationToken = default(CancellationToken))
+    {
+      var rowsAffected = 0;
+      await connection.OpenAsync(cancellationToken);
```

Comment:
Thanks for the tip.  Next commit will implement this library-wide on awaited calls